pkg/thing/delivery/amqp/client.go

```diff
@@ -26,6 +27,7 @@ type msgClientPublisher struct {
 // ClientPublisher is the interface with methods that the publisher should have
 type ClientPublisher interface {
 	SendRegisterDevice(network.RegisterResponseMsg) error
+	SendUnregisterDevice(thingID string, errMsg *string) error
```

Comment:
I would prefer to rename this method to `SendUnregisteredDevice` because we are sending an `unregistered` event. `SendRegisterDevice` should be renamed too, IMO.

Reply:
You are right @joaoaneto! Fixing it the `unregister` for now, in another patch I'll fix for `register` too. Done!